### PR TITLE
Feat/assemble

### PR DESCRIPTION
**Pull request by aNebula**
`trestle assemble` implemented with UT

## Types of changes

- \[ \] Bug fix (non-breaking change which fixes an issue)
- \[x\] New feature (non-breaking change which adds functionality)
- \[ \] Breaking change (fix or feature that would cause existing functionality to change)
- \[x\] My code follows the code style of this project.
- \[x\] My change requires a change to the documentation.
- \[x\] I have updated the documentation accordingly.
- \[x\] I have added tests to cover my changes.
- \[x\] All new and existing tests passed.
- \[x\] All commits are signed-off.

### CODE REVIEW

File: tests/trestle/core/commands/assemble_test.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# limitations under the License.
+"""Tests for trestle assemble command."""
+import pathlib
+import shutil
+import sys
+from unittest import mock
+
+from trestle.cli import Trestle
+from trestle.oscal.catalog import Catalog
+from trestle.utils.load_distributed import load_distributed
```
Comment (butler54) [inline tests/trestle/core/commands/assemble_test.py:24]: So food for thought - I wonder whether with some of these modules we should expose interior functions up to the parent module level. 

e.g. `trestle.utils.load_distributed` lets discuss at some point (no need to change here.)

File: trestle/core/commands/assemble.py [hunk 2]
```diff
@@ -15,10 +15,189 @@
...
+from trestle.core.models.file_content_type import FileContentType
+from trestle.core.models.plans import Plan
+from trestle.oscal import assessment_plan
+from trestle.oscal import assessment_results
+from trestle.oscal import catalog
+from trestle.oscal import component
+from trestle.oscal import poam
+from trestle.oscal import profile
+from trestle.oscal import ssp
+from trestle.oscal import target
```
Comment (butler54) [inline trestle/core/commands/assemble.py:37]: Giveen there is a common pattern of importing everything we should find a simpler way to do this (it should be viable in one line).

**Review by butler54 — CHANGES_REQUESTED** [review]
Mostly good please:
1) Review comments for 'nice to haves'
2) Complete UT coverage. It looks like we are not covering exception criteria.
3) For public docstrings can you please start documenting args in the style here: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

File: tests/trestle/core/commands/assemble_test.py (hunk 1 above)
Comment (aNebula) [reply tests/trestle/core/commands/assemble_test.py:28]: I think it'll feel very satisfying to import modules that way!

File: trestle/core/commands/assemble.py (hunk 2 above)
Comment (aNebula) [reply trestle/core/commands/assemble.py:37]: agreed. We now have 3 commands that do this.

**Review by butler54 — APPROVED** [review]
LGTM

**Pull request merged**